asmcli/components/ca/managed-cas.sh

```diff
@@ -70,9 +70,20 @@ x_enable_workload_certificate_on_membership() {
     }
   }"
 
-  curl -H "Authorization: Bearer ${AUTHTOKEN}" \
-     -X PATCH -H "Content-Type: application/json" -H "Accept: application/json" \
-     -d "${ENABLEFEATURE}" "https://${GKEHUB_API}/v1alpha/projects/${FLEET_ID}/locations/global/features/workloadcertificate?update_mask=membership_specs"
+  # The enablement sometimes fails with an error (an example error is attached below).
+  # Therefore, the enablement is conducted with retries.
+  # "error": {
+  #    "code": 429,
+  #    "message": "Too many requests are currently being executed, try again later.",
+  #    "status": "RESOURCE_EXHAUSTED",
+  #    "details": [...]
+  info "The enablement may take a few minutes ..."
+  for i in {1..3}; do
```

Comment:
could we change this to "retry 3 curl ....". The retry function only tries on failure. This will try repeatedly resulting in unwanted error messages.

Reply:
Do you have suggestion on how to check for success? The current implementation is simple, i.e., just repeating the run three times with 1 minute interval. The retry output is as follows, which does not result in error messages.

asmcli: The enablement may take a few minutes ...
{
  "name": "projects/lt-wip-production-6-23-2022/locations/global/operations/operation-1658433071233-5e4560a1838b4-9a95f898-dcc893c9",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.gkehub.v1alpha.OperationMetadata",
    "createTime": "2022-07-21T19:51:11.277876839Z",
    "target": "projects/lt-wip-production-6-23-2022/locations/global/features/workloadcertificate",
    "verb": "update",
    "cancelRequested": false,
    "apiVersion": "v1alpha"
  },
  "done": false
}
{
  "name": "projects/lt-wip-production-6-23-2022/locations/global/operations/operation-1658433131459-5e4560daf31e2-3186a5ad-c8fb5799",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.gkehub.v1alpha.OperationMetadata",
    "createTime": "2022-07-21T19:52:11.494718928Z",
    "target": "projects/lt-wip-production-6-23-2022/locations/global/features/workloadcertificate",
    "verb": "update",
    "cancelRequested": false,
    "apiVersion": "v1alpha"
  },
  "done": false
}
{
  "name": "projects/lt-wip-production-6-23-2022/locations/global/operations/operation-1658433191860-5e4561148d8a7-1e2b154d-c91836d9",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.gkehub.v1alpha.OperationMetadata",
    "createTime": "2022-07-21T19:53:11.911960371Z",
    "target": "projects/lt-wip-production-6-23-2022/locations/global/features/workloadcertificate",
    "verb": "update",
    "cancelRequested": false,
    "apiVersion": "v1alpha"
  },
  "done": false
}

Reply:
wanted to avoid the messages being repeated 3 times, but isn't that big a deal